library/core/src/array/mod.rs

```diff
@@ -476,6 +476,7 @@ impl<T, const N: usize> [T; N] {
 
     /// Returns a slice containing the entire array. Equivalent to `&s[..]`.
     #[stable(feature = "array_as_slice", since = "1.57.0")]
+    #[rustc_const_stable(feature = "array_as_slice", since = "1.57.0")]
```

Comment:
Shouldn't these be `rustc_const_unstable`? I presume they are not currently callable from stable code since they do not have a `rustc_const_stable` attribute.

Reply:
They **are** callable: [playground](https://play.rust-lang.org/?version=beta&mode=debug&edition=2021&gist=da2151dd0b70ba89945f72f666e6d157) (beta 1.57)

Reply:
My understanding is any stable item that is `const` is also const-stable unless explicitly declared otherwise. The playground link seems to confirm this.

Reply:
Uh, okay, so this is exactly the accidental stabilization @oli-obk was worried about.

Once this PR is in, we should also change that logic, basically treating an absence of a const stability attribute as equal to `rustc_const_unstable`. I am surprised that is not how it works.

Reply:
This PR will actually require a const stability attribute. I'm pretty sure this was _supposed_ to be the case; it was just missing a call for associated functions.

Note that, as written, this doesn't cover const trait impls because there are other checks currently in place (namely that const trait impls can't be stable at all).

Reply:
I understand, but I still think the check should *also* be changed -- at least if that is easy to do, and I imagine it is. The code has to check for the existence of the attribute, and it should 'just' treat non-existence as equivalent to `rustc_const_unstable`.

Call it defense in depth, if you will. ;)

Reply:
Sure, that makes sense.